### PR TITLE
Changed Demography section of tutorial.

### PR DESCRIPTION
**Pull request by gtsambos**
Mostly this is just porting over the SMBE speciation notebooks into the Demography section of the tutorial, but I've also restructured a few other things.

We can talk about this in the tskit meeting on Thursday, but I'm still really not a fan of the OOA example being in this tutorial -- I don't think it highlights any specific feature of msprime particularly clearly, and it is so long and overwhelming that I'm not sure it adds much here. I can see that Aaron has just done some work to revamp it so I'm sorry I didn't raise this earlier 🤦  I think it would work better in some separate 'gallery' of examples, like what we have discussed in the calls. Anyway, for now I have removed it from the 'Demography' section and put it in a ragtag section at the end, for us to think about later.

ping @agladstein and @dvukcevic, perhaps @petrelharp also

### CODE REVIEW

**Comment by jeromekelleher** [comment]
Looks good, thanks @gtsambos. I'm tending to agree with you on the OOA example, especially since it was wrong for several years! I don't think @apragsdale will mind if we remove it, we're planning on using his nice plots upstream in stdpopsim anyway (which is where this belongs).

Something to discuss on Wednesday anyway.

**Comment by apragsdale** [comment]
I can see both ways, but I do think having the OOA model in the tutorial is really useful still. It's one of the more widely used models in simulating human demography, so a lot of people will want to implement it. Having it up front in the demography section of the tutorial lets users run it without worrying about bugs they might have written in (I know it's in stdpopsim, but I don't think msprime users should necessarily need to grow their environment to run it). It's also a model that hits the sweet spot of complexity, to highlight how to model pop structure, changes in migration rates, etc, but not so big that it's pages and pages long. So I think having it accessible and obvious would still be good.

**Comment by molpopgen** [comment]
I second @apragsdale's comment.

**Comment by jeromekelleher** [comment]
OK thanks @apragsdale and @molpopgen, food for discussion on Wednesday!

**Comment by gtsambos** [comment]
Thanks @apragsdale and @molpopgen, I have a few other things to mention about this, but I'll save it for the Wednesday meeting!

Thanks for looking over @agladstein -- I'll make these changes you've suggested and will ping once done.

File: docs/tutorial.rst [hunk 1]
```diff
@@ -553,251 +560,318 @@ scikit-allel haplotype array.
...
+To understand why, recall that ``msprime`` is a `coalescent`-based simulator.
+The simulation will run backwards-in-time, simulating until `all` samples have
+coalesced to a single common ancestor at each genomic location.
+However, with no migration between our two populations, samples in one
+population will never coalesce with samples in another population.
+To fix this, let's add some migration events to the specific demographic history.
 
 **********
-Demography
+Migrations
```
[the remark this answers is not in the document]
Comment (gtsambos) [reply docs/tutorial.rst:623]: Hmm, I've just realised that to do this change

> mass migration
> use case for divergence
> use case for admixture
> migration
> without mass migration
> with mass migration (talk about potential issue)

I'm going to have to change the tex pictures in addition to the text, which will take a bit more time as I'm going to have to track down the original .tex files and remind myself of how to code in TikZ (they aren't in the SMBE repo, oops :/)

Would it be okay if we perhaps pulled this request in as is, and changed the order of the migration subsections a bit later on? Just so this pull request isn't hanging around for too long

@agladstein -- I am just about to push up a commit with the more minor fixes you've suggested above

**Comment by gtsambos** [comment]
Hmm, I've just realised that to do this change

> mass migration
> use case for divergence
> use case for admixture
> migration
> without mass migration
> with mass migration (talk about potential issue)

I'm going to have to change the tex pictures in addition to the text, which will take a bit more time as I'm going to have to track down the original .tex files and remind myself of how to code in TikZ (they aren't in the SMBE repo, oops :/)

Would it be okay if we perhaps pulled this request in as is, and changed the order of the migration subsections a bit later on? Just so this pull request isn't hanging around for too long

@agladstein -- I am just about to push up a commit with the more minor fixes you've suggested above

**Comment by agladstein** [comment]
Yes, I did realize that. That's fine!

**Comment by petrelharp** [comment]
>  I'm tending to agree with you on the OOA example, especially since it was wrong for several years! 

I think *because* it was wrong we should leave it in, so that we can have a warning note saying that it was wrong before? But, I agree that having it in a "gallery" is a good idea.

**Comment by jeromekelleher** [comment]
> I think because it was wrong we should leave it in, so that we can have a warning note saying that it was wrong before? But, I agree that having it in a "gallery" is a good idea.

I'm easy either way - let's see what people think tomorrow on the call.

**Comment by gtsambos** [comment]
Okay, cool. Unless there's more to change in this PR, it might be ready to pull in then?

**Comment by jeromekelleher** [comment]
Great, thanks @gtsambos. I rebased this and added back in Aaron's nice OOA picture so we don't forget about it when we're repurposing the OOA example.

**Comment by gtsambos** [comment]
Thanks, good spot @jeromekelleher -- I must have accidentally taken out the picture when fixing a local merge conflict. We'll chat soon.

**Pull request merged**